### PR TITLE
WIP: Api pagination

### PR DESCRIPTION
**Pull request by bbujisic**

### CODE REVIEW

File: src/DataStructure/Collection.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Platformsh\Client\DataStructure;
```
Comment (pjcdawkins) [inline src/DataStructure/Collection.php:3]: Hmmm I intended `DataStructure` for very abstract things (it was for `ReadOnlyStructureTrait`), and this class is api-specific.

File: src/Model/ApiResourceBase.php [hunk 2]
```diff
@@ -153,52 +160,64 @@ public function ensureFull()
...
 
+    /**
+     * Get a collection of resources.
+     *
+     * @param PlatformClient $client  A suitably configured Platform client.
+     * @param QueryInterface $query   An instance of query interface. It will be used to build a guzzle query.
+     *
+     * @return Collection;
+     */
+    public static function getCollection(PlatformClient $client, ?QueryInterface $query = null)
```
Comment (pjcdawkins) [inline src/Model/ApiResourceBase.php:195]: Presumably it doesn't need the `= null` if it already has a nullable type hint

File: src/DataStructure/Collection.php (hunk 1 above)
Comment (bbujisic) [reply src/DataStructure/Collection.php:3]: I'm pretty bad at naming things. Ideas?

File: src/Model/ApiResourceBase.php (hunk 2 above)
Comment (bbujisic) [reply src/Model/ApiResourceBase.php:195]: Unfortunately, it does need it. It returns a "Too few arguments" error unless `null` explicitly provided.

**Pull request closed without merging**